pkg/blobinfocache/sqlite/sqlite.go

```diff
@@ -467,10 +470,33 @@ func (sqc *cache) appendReplacementCandidates(candidates []prioritize.CandidateW
 	if err := rows.Err(); err != nil {
 		return nil, fmt.Errorf("iterating through locations: %w", err)
 	}
+
+	if len(res) == 0 && v2Output {
+		compressorName := blobinfocache.UnknownCompression
+		compressor, found, err := querySingleValue[string](tx, "SELECT compressor FROM DigestCompressors WHERE digest = ?", digest.String())
+		if err != nil {
+			return nil, fmt.Errorf("scanning compressorName: %w", err)
+		}
+		if found {
+			compressorName = compressor
+		}
+		if compressorName != blobinfocache.UnknownCompression {
```

Comment:
(`DigestCompressors` are documented to not contain `UnknownCompression` — and the code just above relies on that in the `v2Output` case — so this could be
```go
compressorName, found, err := …
if err …
if found { /* use compressorName */ }
```
)